### DISCUSSION TITLE
thoughts about `pacman --groups`

### DISCUSSION THREAD

**Discussion by goyalyashpal** [msys2/MSYS2-packages#3550] [Q&A] [answered]
```console
$ pacman -Qh
  -g, --groups         view all members of a package group
  -l, --list           list the files owned by the queried package

$ pacman -Sh
  -g, --groups         view all members of a package group
                       (-gg to view all groups and members)
```

- `-Qgg` is valid too, t'd be nice to have that line in the `-Qh` help (as it is in `-Sh`)

### Some other miscellaneous thoughts about the `-g` in `pacman`

`-gg` show groups _and_ members 
- is there any way to list these in a tree view say with `pactree` smth? 
(would help with easy separation of different groups)
- any way to list only groups, not their members?
- and some `--info` like view for groups? 

```console
$ pacman -Sig editors
error: invalid option: '--info' and '--groups' may not be used together
```

### Details on question about the `--info` view

like showing the details of a group in the way details are shown for packages?

At least following 3 fields maybe relevant: 
"Provides" (for members), "Download Size", "Installed Size"

```console
$ pacman -Si nnn
Groups          : None
Provides        : None
Depends On      : bash  ncurses  libreadline
Optional Deps   : atool: for more archive formats
                  libarchive: for more archive formats
                  zip: for zip archive format
                  unzip: for zip archive format
                  rclone: mount remotes
Download Size   : 63.35 KiB
Installed Size  : 114.17 KiB

$ pacman -Sg editors
editors emacs
editors nano
editors vim
```

Imagined is smth like:
```console
$ pacman -Sig editors
Provides        : emacs  nano  vim
Download Size   : 63.35 KiB
Installed Size  : 114.17 KiB
```

----
### Tangentials:

- What are other packages and tools for aiding with `pacman`
i know only 2 so far: `pactree` and <tt>[pacboy]</tt> (from `pactoys`)
- I call the view used in `--info`  to be "form" or "semi-tabular" view (contrast with "linear" or "tree" view etc from pactree) 
is there a technical name for this view? or what will u call this? 😅 
**Edit:** Yes, [two-pane] view i guess.

[pacboy]: https://github.com/msys2/MSYS2-packages/issues/3457#issuecomment-1366980203
[two-pane]: https://wiki.archlinux.org/title/pacman/Tips_and_tricks#Browsing_packages

**Comment by lazka** [comment, marked as answer]
(Note that we don't develop pacman, it's developed by Arch Linux: https://gitlab.archlinux.org/pacman/pacman)

> * is there any way to list these in a tree view say with `pactree` smth?
>   (would help with easy separation of different groups)
> 
> * any way to list only groups, not their members?

You could do `pacman -Qg | cut -d" " -f1 | sort | uniq`

> * and some `--info` like view for groups?

I don't think so.

pacman mainly operates on packages, and groups are just a field of packages.

See:

* https://man.archlinux.org/man/pacman.8
* https://wiki.archlinux.org/title/pacman
* https://wiki.archlinux.org/title/pacman/Tips_and_tricks

**Comment by goyalyashpal** [comment]
As for the other tools thing:

- this "[updating]" page (msys2) provides several commands - several of which i _think_ are provided within `pacman` itself
`pacdiff` , `paccache`, `pacman-key`
- this above pacman [tips/tricks] (arch) lists some more:
`expac`, `pacgraph`, `pacolog`, `pakbak-git`
`pacutils` (`pacreport` , `paccheck`), 
`powerpill`
- [wiki/pacman]: `pacman-contrib` contains various tools too


[updating]: https://www.msys2.org/docs/updating/
[tips/tricks]: https://wiki.archlinux.org/title/pacman/Tips_and_tricks
[wiki/pacman]: https://wiki.archlinux.org/title/Pacman#top-page


```console
$ pacman -Ss man
msys/namcap 3.3.1-4
    A Pacman package analyzer

<other interesting ones:>
mingw64/mingw-w64-x86_64-winico 0.6-3
    Tk extension for Windows for enhanced icon handling and manipulation of an icon in the Windows taskbar and system tray (mingw-w64)
mingw64/mingw-w64-x86_64-wxsvg-wx3.2 1.5.24-1
    A C++ library to create, manipulate and render SVG files (mingw-w64)
mingw64/mingw-w64-x86_64-xmlstarlet 1.6.1-1
    Command-line XML toolkit
ucrt64/mingw-w64-ucrt-x86_64-hexyl 0.12.0-1
    A command-line hex viewer
ucrt64/mingw-w64-ucrt-x86_64-highs 1.4.2-1
    High performance linear optimization software (mingw-w64)
ucrt64/mingw-w64-ucrt-x86_64-hyperfine 1.15.0-1
    A command-line benchmarking tool (mingw-w64)
ucrt64/mingw-w64-ucrt-x86_64-imagemagick 7.1.0.62-1
    An image viewing/manipulation program (mingw-w64)
ucrt64/mingw-w64-ucrt-x86_64-irrlicht 1.8.5-1
    An open source high performance realtime 3D graphics engine (mingw-w64)
ucrt64/mingw-w64-ucrt-x86_64-libunistring 1.1-1
    Library for manipulating Unicode strings and C strings. (mingw-w64)
ucrt64/mingw-w64-ucrt-x86_64-lldb 15.0.7-2
    Next generation, high-performance debugger (mingw-w64)
ucrt64/mingw-w64-ucrt-x86_64-lsd 0.23.1-1
    The next gen ls command (mingw-w64)
ucrt64/mingw-w64-ucrt-x86_64-manim 0.17.2-1
    Animation engine for explanatory math videos (community edition) (mingw-w64)
ucrt64/mingw-w64-ucrt-x86_64-mimalloc 2.0.9-1
    General-purpose allocator with excellent performance characteristics (mingw-w64)
msys/axel 2.17.11-2
    Light command line download accelerator for Linux and Unix

```
